File: selfstudy/js/cities_filterhigh.js
/*
task 75:
Cities - FilterHigh
You need to return only the values greater than 50 from city populations in thousands
*/

function above_fifty_cities(populations) {
  return populations.filter(x => x > 50);
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: above_fifty_cities([10, 20, 30]) },
  { input: [50, 60, 70], expected: above_fifty_cities([50, 60, 70]) },
  { input: [1, 2, 3], expected: above_fifty_cities([1, 2, 3]) },
  { input: [100, 200], expected: above_fifty_cities([100, 200]) },
  { input: [], expected: above_fifty_cities([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('above_fifty_cities', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(above_fifty_cities(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, above_fifty_cities(t.input) === t.expected || JSON.stringify(above_fifty_cities(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// done

File: selfstudy/golang/cities_filterhigh.go
/*
task 275:
Cities - FilterHigh
You need to return only the values greater than 50 from city populations in thousands
*/

package main

import (
	"fmt"
	"reflect"
)

func Above_fiftyCities(populations []int) []int {
	var res []int
	for _, v := range populations {
		if v > 50 { res = append(res, v) }
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Above_fiftyCities([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Above_fiftyCities([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Above_fiftyCities([]int{1, 2, 3})},
		{[]int{100, 200}, Above_fiftyCities([]int{100, 200})},
		{[]int{}, Above_fiftyCities([]int{})},
	}
	
	for idx, t := range tests {
		res := Above_fiftyCities(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done
